LanguageFeatures/Augmentation-libraries/augmented_expression_A05_t25_lib.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting operators: When augmenting an operator, `augmented` must be
+///   followed by the operator. For example when augmenting `+` you must do
+///   `augmented + 1`, and when augmenting `[]` you must do `augmented[<arg>]`.
+///   These constructs invoke the augmented operator, and are the only valid
+///   uses of `augmented` in these contexts.
+///
+/// @description Checks that it is a compile-time error to use a type whose name
+/// is `augmented` in `is` and `as` expressions.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmented_expression_A05_t25.dart';
+
+augment class C {
+  augment String operator +(Object? other) {
+    print(null as augmented);
+//                ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    print(null is augmented);
+//                ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    return "";
+  }
+}
+
+augment mixin M {
+  augment String operator +(Object? other) {
+    print(null as augmented);
+//                ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    print(null is augmented);
+//                ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    return "";
+  }
+}
+
+augment enum E {
+  augment e1;
+  augment String operator +(Object? other) {
+    print(null as augmented);
+//                ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    print(null is augmented);
+//                ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    return "";
+  }
+}
+
+augment extension Ext {
+  augment String operator +(Object? other) {
+    switch ("") {
```

Comment:
Yes. Using copy-paste for writing test has some side-effects :) Fixed.

Reply:
Oh, was it fixed, though?

Reply:
Yes, it was fixed

Reply:
Oh, but I thought there shouldn't be an occurrence of the word 'switch' anywhere in [LanguageFeatures/Augmentation-libraries/augmented_expression_A05_t25_lib.dart](https://github.com/dart-lang/co19/pull/2766/files/4938aa0acf9dc6c5ad7629af35b98cb51017e021#diff-bf241a0efe38fba440e79f3bb9704c7f610d9778db2f2c40a67f46ddc58917b9) because this is a test about `is` and `as`. So there's something I missed?

Reply:
I'm sorry, when I click on the link at the top of this box I'll see the _old_ version of the file augmented_expression_A05_t25_lib.dart, not the most recent one. I thought I was looking at the most recent one. 

OK, landing!